ocaml/xapi/test_vdi_allowed_operations.ml

```diff
@@ -262,6 +262,47 @@ let test_cbt =
       )
   in
 
+
+  let test_vdi_data_destroy () =
+    let __context = Mock.make_context_with_new_db "Mock context" in
+
+    (* change VDI properties so that data_destroy passes *)
+    let pass_data_destroy vdi =
+      let sr = Db.VDI.get_SR ~__context ~self:vdi in
+      Db.SR.set_is_tools_sr ~__context ~self:sr ~value:false;
+      Db.VDI.set_is_a_snapshot ~__context ~self:vdi ~value:true;
+      Db.VDI.set_managed ~__context ~self:vdi ~value:true;
+      Db.VDI.set_cbt_enabled ~__context ~self:vdi ~value:true in
+
+    (* change VDI properties one by one so data_destroy only fails for the indicated reason *)
+    List.iter (fun (vdi_fun, api_error) ->
+        run_assert_equal_with_vdi ~__context ~vdi_fun `data_destroy (api_error))
+
+      [
+        (* ensure VDI.data_destroy works before introducing errors *)
+        (fun vdi -> pass_data_destroy vdi;
+        ) ,         None ;
```

Comment:
Ooops sorry, I forgot one test, we should also check that data_destroy is actually idempotent: that it does not return an error for a VDI of type `cbt_metadata`.